File: 948-sort-an-array/sort-an-array.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        def merge_sort(arr):
            if len(arr) > 1:
                mid = len(arr)//2
                left_value = merge_sort(arr[:mid])
                right_value = merge_sort(arr[mid:])

                l = 0
                r = 0
                k = 0
                ans = []
                while(l < len(left_value) and r < len(right_value)):
                    if left_value[l] <= right_value[r]:
                        arr[k] = left_value[l]
                        l += 1
                    else:
                        arr[k] = right_value[r]
                        r += 1
                    k += 1
                while(l < len(left_value) and k < len(arr)):
                    arr[k] = left_value[l]
                    l += 1
                    k += 1
                while(r < len(right_value) and k < len(arr)):
                    arr[k] = right_value[r]
                    r += 1
                    k += 1
                return arr
            else:
                return arr
        merge_sort(nums)
        return nums

File: 948-sort-an-array/sort-an-array.go
import "fmt"
func merge_sort(n []int ) []int{
	if len(n) > 1{
		mid := len(n) / 2
		left := make([]int, mid)
		right := make([]int, len(n)-mid)
		copy(left, n[:mid])
		copy(right, n[mid:])
        merge_sort(left)
		merge_sort(right)
		var l int= 0
		var r int = 0
		var i int = 0
		for l < len(left) && r < len(right){
			if left[l] <= right[r]{
				n[i] = left[l]
				l += 1
			} else {
				n[i] = right[r]
				r += 1
			}
			i += 1
		}
		for l < len(left){
			n[i] = left[l]
			i += 1
			l += 1
		}
		for r < len(right) {
			n[i] = right[r]
			i += 1
			r += 1
		}
		
		
		return n
	}
	return n
}
func sortArray(nums []int) []int {
    nums = merge_sort(nums)
    return nums
    
}